### PR TITLE
fix(app): cautiously access values from results of commands in case of analysis failure

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Protect against accessing things off of `command.result` that may not be there.  The typescript types for all `RunTimeCommand`s have been updated to contain an optional result key. This maps to the actual backend behavior, and should surface any places in the where we are not respecting the possibility of an absent result. I've addressed what I believe are all outstanding locations in the app where we were parsing information out of the result key without protection for the nullish case. 

# Review requests

Check code changes, and smoke test uploading a protocol.  Ideally one that will have a command like `loadLabware` fail where a result would be present if it succeeded.  

# Risk assessment
low, this shouldn't effect normal behavior of the app, only prevent bugs

### CODE REVIEW

File: api-client/src/protocols/utils.ts [hunk 1]
```diff
@@ -2,13 +2,14 @@
 import reduce from 'lodash/reduce'
 
 import { COLORS } from '@opentrons/components/src/ui-style-constants'
-import type {
+import {
```
Comment (shlokamin) [inline api-client/src/protocols/utils.ts]: nit: might be worth keeping the `type` prefix and adding a new import statement for `getLabwareDefURI`

File: api-client/src/protocols/utils.ts [hunk 2]
```diff
@@ -53,14 +54,14 @@ export function parsePipetteEntity(
   )
   if (rightPipetteCommand != null) {
     pipetteEntity.push({
-      id: rightPipetteCommand.result.pipetteId,
+      id: rightPipetteCommand.result?.pipetteId ?? '',
```
Comment (shlokamin) [inline api-client/src/protocols/utils.ts:57]: falling back to empty string seems like the most reasonable thing go me. i would think that consumers shouldnt be calling this util at all if analysis has failed though — we'd be providing info that is technically invalid

Comment (shlokamin) [reply api-client/src/protocols/utils.ts:57]: so i guess the broader question im trying to ask is: if analysis has failed, should we be trying to access things like pipette entities?

File: app/src/molecules/DeckThumbnail/index.tsx [hunk 3]
```diff
@@ -69,16 +69,18 @@ export function DeckThumbnail(props: DeckThumbnailProps): JSX.Element {
               ? initialLoadedLabwareBySlot[slotId]
               : null
           const labwareInModule =
-            moduleInSlot &&
+            moduleInSlot?.result?.moduleId != null &&
             moduleInSlot.result.moduleId in initialLoadedLabwareByModuleId
-              ? initialLoadedLabwareByModuleId[moduleInSlot.result.moduleId]
+              ? initialLoadedLabwareByModuleId[moduleInSlot.result?.moduleId]
```
Comment (shlokamin) [inline app/src/molecules/DeckThumbnail/index.tsx]: nit: you already type guarded in the and statement above so i don't think you need the question mark

```suggestion
              ? initialLoadedLabwareByModuleId[moduleInSlot.result.moduleId]
```

File: app/src/molecules/PythonLabwareOffsetSnippet/createSnippet.ts [hunk 4]
```diff
@@ -88,7 +90,7 @@ export function createSnippet(
         }`
         moduleVariableById = {
           ...moduleVariableById,
-          [command.result.moduleId]: moduleVariable,
+          [command.result?.moduleId ?? '']: moduleVariable,
```
Comment (shlokamin) [inline app/src/molecules/PythonLabwareOffsetSnippet/createSnippet.ts:93]: same thing here for empty string fallbacks — seems like the most reasonable thing but it feels like consumers should not be calling this function with failed analysis commands. so in the future we could consider typing the input to only take in a run time command interfaces with results keys. 

thats a bigger discussion tho and we can bring it up later with RSS

**Review by shlokamin — APPROVED** [review]
LGTM! deleted the results key in load labware commands in a local protocol analysis file and nothing bad seems to be happening!

File: api-client/src/protocols/utils.ts (hunk 2 above)
Comment (b-cooper) [reply api-client/src/protocols/utils.ts:57]: That's a good question.  Ideally we wouldn't be accessing anything off of a failed analysis. Though there may be some value in viewing the information that was parseable, even if an entire protocol couldn't analyze successfully

**Pull request merged**